src/github.rs

```diff
@@ -1076,6 +1076,80 @@ impl Issue {
             .await?;
         Ok(())
     }
+
+    /// Checks if the PR's parent commit is too old.
+    ///
+    /// This determines if a PR needs updating by examining the first parent of the PR's head commit,
+    /// which typically represents the base branch commit that the PR is based on.
+    ///
+    /// If this parent commit is older than the specified threshold, it suggests the PR
+    /// should be updated/rebased to a more recent version of the base branch.
+    ///
+    /// Returns:
+    /// - Ok(Some(days_old)) - If parent commit is older than the threshold
+    /// - Ok(None) - If not a PR, can't get commit details, or parent is within threshold
+    /// - Err(...) - If an error occurred during processing
+    pub async fn is_parent_commit_too_old(
+        &self,
+        client: &GithubClient,
+        max_days_old: usize,
+    ) -> anyhow::Result<Option<usize>> {
+        if !self.is_pr() {
+            return Ok(None);
+        }
+
+        // Get the head commit SHA
+        let Some(head) = &self.head else {
+            return Ok(None);
+        };
+
+        // Get the commit details
+        let commit_url = format!("{}/commits/{}", head.repo.url(client), head.sha);
+        let commit: GithubCommit = match client.json(client.get(&commit_url)).await {
+            Ok(commit) => commit,
+            Err(e) => {
+                log::warn!(
+                    "Failed to get commit details for PR #{}: {}",
+                    self.number,
+                    e
+                );
+                return Ok(None);
+            }
+        };
+
+        // Get the first parent (it should be from the base branch)
+        if commit.parents.is_empty() {
+            return Ok(None);
+        }
+
+        // The first parent should be from the base branch
+        let parent_sha = &commit.parents[0].sha;
+
+        // Get the parent commit details to check its date
+        let parent_url = format!("{}/commits/{}", head.repo.url(client), parent_sha);
+        let parent_commit: GithubCommit = match client.json(client.get(&parent_url)).await {
+            Ok(commit) => commit,
+            Err(e) => {
+                log::warn!(
+                    "Failed to get parent commit details for PR #{}: {}",
+                    self.number,
+                    e
+                );
+                return Ok(None);
+            }
+        };
+
+        // Check if the parent commit is too old
+        let parent_date = parent_commit.commit.author.date;
```

Comment:
Let's put this logic into a separate function, and only return the author date of the base branch commit from this function, to avoid it doing too many things.